cpp/include/cudf/copying.hpp

```diff
@@ -44,6 +44,17 @@ gdf_column empty_like(gdf_column const& input);
  */
 gdf_column allocate_like(gdf_column const& input, bool allocate_mask_if_exists = true, cudaStream_t stream = 0);
 
+/**
+ * @brief Allocates a new column of the same size as the input with the specified type.
+ *
+ * @param input The input column to emulate
+ * @param output_type The output type
+ * @param allocate_mask_if_exists Optional whether or not to allocate bitmask if it exists in input
+ * @param stream Optional stream in which to perform copies
+ * @return gdf_column An allocated column of same size and type of input
+ */
+gdf_column allocate_size_dtype(gdf_size_type size, gdf_dtype output_type, bool allocate_mask=true, cudaStream_t stream=0);
```

Comment:
How about just use this function that @devavret created here: https://github.com/rapidsai/cudf/pull/2202/files#diff-85c936e50a5ef48c663a0124b7aeddceR57

I think that naming is more intuitive as well.

Reply:
@harrism @jrhemstad I am not sure what is the conclusion here?

Reply:
Since you don't need anything from another column, this isn't a copy, it's allocation from scratch. So I agree with Jake. However in @devavret's PR I asked him to move his `allocate_column` function out of copying (since it's not in any way a copy).

Reply:
Okay I will copy from his PR and put it under `/legacy/column.*`. The function is short enough so it shouldn't cause too much trouble when merging.

Reply:
Devavret moved it under legacy in his PR.